### PR TITLE
Framework callback for engine memory pressure events

### PR DESCRIPTION
**Pull request by jason-simmons**

### CODE REVIEW

**Comment by jason-simmons** [comment]
@abarth

**Comment by Hixie** [comment]
test?

**Comment by Hixie** [comment]
LGTM. It's likely we'll eventually move this up to the services binding so that we can make use of it in the Cache logic, but that's a problem for a later day and would be backwards compatible anyway.

**Comment by jason-simmons** [comment]
Added a test for WidgetsBinding

File: packages/flutter/lib/src/widgets/app.dart [hunk 1]
```diff
@@ -180,6 +180,9 @@ class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserv
   void didChangeAppLifecycleState(AppLifecycleState state) { }
 
   @override
+  void didHaveMemoryPressure() { }
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/app.dart]: No need for this override.  The default do-nothing implementation is fine.

File: packages/flutter/lib/src/widgets/app.dart [hunk 2]
```diff
@@ -180,6 +180,9 @@ class _WidgetsAppState extends State<WidgetsApp> implements WidgetsBindingObserv
   void didChangeAppLifecycleState(AppLifecycleState state) { }
```
Comment (abarth) [inline packages/flutter/lib/src/widgets/app.dart]: You can also remove this one.

File: packages/flutter/lib/src/widgets/app.dart (hunk 1 above)
Comment (jason-simmons) [inline packages/flutter/lib/src/widgets/app.dart]: The analyzer warns if these overrides are not provided

Comment (abarth) [inline packages/flutter/lib/src/widgets/app.dart]: Oh, I see.  Because you extend State but only implement observer.  ok.

**Pull request merged**